tests/unit/rankers/test_algorithms.py

```diff
@@ -288,42 +293,50 @@ def test_personalize_topic_limit(self):
                                for t in full_topic_profile.curator_topics
                                if t.curator_topic_label in input_topics]
 
-        for test_limit in [1, 3, 5]:
-            output_configs = personalize_topic_slates(input_configs, full_topic_profile, test_limit)
+        for test_limit, topic_ranker in zip([1, 3], [top1_topics, top3_topics]):
+            output_configs = topic_ranker(input_configs, full_topic_profile)
             ordered_output_topics = [c.curator_topic_label for c in output_configs]
             print(len(output_configs), test_limit)
             assert len(output_configs) == test_limit
             assert ordered_output_topics == personalized_topics[:test_limit]
 
-    def test_android_discover(self):
+    def test_hybrid_lineups(self):
         ''' this test is the case where one topic slate is returned from a
         lineup with a mix of curated and uncurated slates
         '''
         full_topic_profile = self._read_json_asset("recit_full_user_profile.json")
-        input_configs = generate_hybrid_configs()
-        input_topics = [c.curator_topic_label for c in input_configs]
-        # these are already sorted by RecIt
-        personalizable_slates = [t.curator_topic_label
-                                 for t in full_topic_profile.curator_topics
-                                 if t.curator_topic_label in input_topics]
-        other_slates = [t.id for t in input_configs
-                        if t.curator_topic_label not in personalizable_slates]
-
-        for topic_limit in range(1, 9):
-            # one non-topic slate should be at start of lineup
-            output_configs = personalize_topic_slates(input_configs, full_topic_profile, topic_limit)
-            ordered_output_topics = [c.curator_topic_label for c in output_configs]
-            assert len(output_configs) == len(other_slates) + topic_limit
-            # first config is not personalziable and should be same as input
-            assert input_configs[0] == output_configs[0]
-            # returned topic slate should be highest ranked
-            assert ordered_output_topics[1:1+topic_limit] == personalizable_slates[:topic_limit]
-            # last two output slates are also not personalizable
-            assert input_configs[-2:] == output_configs[-2:]
-
+        personalized_topics = [x.curator_topic_label for x in full_topic_profile.curator_topics]
+
+        for lineup_id in [ANDROID_DISCOVER_LINEUP_ID, WEB_HOME_LINEUP_ID]:
```

Comment:
Pytest supports [parameterized tests](https://docs.pytest.org/en/6.2.x/example/parametrize.html), which have the benefit of telling you which parameter failed, and keeping tests a little more readable.